### PR TITLE
Add locations endpoint test

### PR DESCRIPTION
**Pull request by jkevinruiz**
Signed-off-by: Kevin Ruiz <jkevincodes@gmail.com>

Contributes to #142 

## What did you do?

Add test for locations endpoint

## Why did you do it?

Missing test

## How have you tested it?

Local test run. In order to run endpoint test currently, we would need to change the following setting in the `settings.py`
`
STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.StaticFilesStorage'
`

Then run the test with `./manage.py test`

```
**Using SQLite3**
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
02-01 02:55:02 add_states:handle [INFO] l31: OH:Ohio found, skipping.
.02-01 02:55:02 add_states:handle [INFO] l31: GA:Georgia missing. Adding now.
02-01 02:55:02 add_states:handle [INFO] l31: MN:Minnesota missing. Adding now.
02-01 02:55:02 add_states:handle [INFO] l31: TX:Texas missing. Adding now.
.02-01 02:55:02 add_states:handle [INFO] l31: CA:California missing. Adding now.
......02-01 02:55:03 views:health [INFO] 177:Health AnonymousUser
.02-01 02:55:03 views:health [INFO] 177:Health AnonymousUser
....
----------------------------------------------------------------------
Ran 13 tests in 2.915s

OK
Destroying test database for alias 'default'...

```

## Were docs updated if needed?

- [X] No
- [ ] Yes
- [ ] N/A

#### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new console logs
- [ ] Fixes entire issue
- [ ] Partial fix for issue

### CODE REVIEW

File: cfc_app/tests.py [hunk 1]
```diff
@@ -35,11 +35,34 @@ def test_health_status_is_up(self):
         self.assertContains(response, '{"status": "UP"}')
 
     def test_health_status_redirects(self):
-        """ Test that '/health' is redirected to '/health/' and returned with RC=301 """
+        """ Test that '/health' is redirected to '/health/' with RC=301 """
 
         response = self.client.get('/health')
-        self.assertEqual(response.status_code, 301)
-        self.assertEqual(response.url, '/health/')
+        self.assertRedirects(response, '/locations/', 301, 200)
```
Comment (jkevinruiz) [inline cfc_app/tests.py:41]: found a redirects assertion, which I think would be better to use for redirects test

File: cfc_app/tests.py [hunk 2]
```diff
@@ -35,11 +35,34 @@ def test_health_status_is_up(self):
...
+    """ Locations Endpoint use to show all supported locations """
+
+    def test_locations_template(self):
+        """ Test that locations uses 'locations.html' template """
+
+        response = self.client.get('/locations/')
+        self.assertTemplateUsed(response, 'locations.html')
+
+    def test_locations_default_locations_loaded(self):
+        """ Test that default locations are loaded"""
```
Comment (jkevinruiz) [inline cfc_app/tests.py:53]: locations endpoint loads default location, not sure if this would be a robust test, any suggestions are welcome

**Comment by jkevinruiz** [comment]
I'm sure there are a couple more test i'm missing, would appreciate some other suggestions on what to test

File: cfc_app/tests.py (hunk 1 above)
Comment (jkevinruiz) [inline cfc_app/tests.py]: found an `assertRedirects` method for testing redirects

**Comment by jkevinruiz** [comment]
@upkarlidder @tonypearson

File: cfc_app/tests.py [hunk 3]
```diff
@@ -35,11 +35,34 @@ def test_health_status_is_up(self):
         self.assertContains(response, '{"status": "UP"}')
 
     def test_health_status_redirects(self):
-        """ Test that '/health' is redirected to '/health/' and returned with RC=301 """
+        """ Test that '/health' is redirected to '/health/' with RC=301 """
 
         response = self.client.get('/health')
-        self.assertEqual(response.status_code, 301)
-        self.assertEqual(response.url, '/health/')
+        self.assertRedirects(response, '/health/', 301, 200)
```
Comment (upkarlidder) [inline cfc_app/tests.py:41]: LGTM.

File: cfc_app/tests.py [hunk 4]
```diff
@@ -35,11 +35,34 @@ def test_health_status_is_up(self):
...
+
+class LocationsEndpointTests(TestCase):
+    """ Locations Endpoint use to show all supported locations """
+
+    def test_locations_template(self):
+        """ Test that locations uses 'locations.html' template """
+
+        response = self.client.get('/locations/')
+        self.assertTemplateUsed(response, 'locations.html')
```
Comment (upkarlidder) [inline cfc_app/tests.py:50]: LGTM

File: cfc_app/tests.py [hunk 5]
```diff
@@ -35,11 +35,34 @@ def test_health_status_is_up(self):
...
+    def test_locations_default_locations_loaded(self):
+        """ Test that default locations are loaded"""
+
+        response = self.client.get('/locations/')
+        self.assertNotContains(response, 'world')
+        self.assertContains(response, 'United States')
+        self.assertContains(response, 'Arizona')
+        self.assertContains(response, 'Ohio')
+
+    def test_locations_redirects(self):
```
Comment (upkarlidder) [inline cfc_app/tests.py:61]: LGTM

File: cfc_app/tests.py (hunk 2 above)
Comment (upkarlidder) [reply cfc_app/tests.py:53]: @jkevinruiz, I had a conversation with @tonypearson. This is a good test to check if the seed data was loaded successfully. However we don't think it should run every time code is committed to the repo. This is a data integrity test and not a unit test. We recommend moving this out of the unit test file and adding usage instructions in the README.md if somebody wanted to test loading the seed data.

**Review by upkarlidder — CHANGES_REQUESTED** [review]
@jkevinruiz pls see review comments.

File: cfc_app/tests.py (hunk 2 above)
Comment (jkevinruiz) [reply cfc_app/tests.py:53]: @upkarlidder that makes sense, i'll remove the default tests and update the readme. I'm guessing the other test works fine. 

Do you have any other test cases we could write around this endpoints? From examples I've seen endpoints are mostly tested with templates and the response data's they return in django.

**Pull request merged**